crates/sui-framework/packages/sui-framework/sources/k_of_n_upgrade_policy.move

```diff
@@ -0,0 +1,522 @@
+        signer: address,
+    }
+
+    /// Record the transfer for a `Ballot`.
+    struct BallotTransfered has copy, drop {
+        /// the ID of the proposal (`ProposedUpgrade` instance)
+        proposal: ID,
+        /// digest of the proposal
+        digest: vector<u8>,
+        /// ballot used for this vote
+        ballot: ID,
+        /// signer of the transfer operation
+        signer: address,
+        /// original owner of the ballot, that is the address the
+        /// ballot was first sent to
+        original_owner: address,
+        /// the receiver of the ballot
+        ballot_receiver: address,
+    }
+
+    /// Record a deleted `Ballot`.
+    struct BallotDeleted has copy, drop {
+        /// the ID of the proposal (`ProposedUpgrade` instance)
+        proposal: ID,
+        /// digest of the proposal
+        digest: vector<u8>,
+        /// ballot used for this vote
+        ballot: ID,
+        /// signer of the deleted operation
+        signer: address,
+        /// original owner of the ballot, that is the address the
+        /// ballot was first sent to
+        original_owner: address,
+    }
+
+    /// Allowed voters must in the [2, 100] range.
+    const EAllowedVotersError: u64 = 0;
+    /// Required votes must be less than allowed voters.
+    const ERequiredVotesError: u64 = 1;
+    /// The signer of the proposed upgrade is not one of the allowed voters.
+    const EInvalidProposer: u64 = 2;
+    /// The `Ballot` used to vote is not for the correct proposal (`ProposedUpgrade`)
+    const EInvalidBallot: u64 = 3;
+    /// Not enough votes to perform the upgrade.
+    const ENotEnoughVotes: u64 = 4;
+    /// An upgrade was issued already, and the operation requested failed.
+    const EAlreadyIssued: u64 = 5;
+    /// The operation required the signer to be the same as the upgrade proposer.
+    const ESignerMismatch: u64 = 6;
+    /// Proposal (`KofNUpgradeCap`) and upgrade (`ProposedUpgrade`) do not match.
+    const EInvalidProposalForUpgrade: u64 = 7;
+
+    /// Create a `KofNUpgradeCap` given an `UpgradeCap`.
+    /// The returned instance is the only and exclusive controller of upgrades. 
+    /// If the transaction sender is not in the set of allowed voters it will be
+    /// added. The sender is the publisher of the package which is always allowed
+    /// to vote.
+    /// The `k` (`required_votes`) out of `n` (length of `voters`) is set up
+    /// at construction time and it is immutable.
+    public fun new(
+        cap: UpgradeCap,
+        required_votes: u64,
+        voters: VecSet<address>,
+        ctx: &mut TxContext,
+    ): KofNUpgradeCap {
+        assert!(vec_set::size(&voters) > 1, EAllowedVotersError);
+        // add the signer to the list of voters if that was not there
+        let signer = tx_context::sender(ctx);
+        if (!vec_set::contains(&voters, &signer)) {
+            vec_set::insert(&mut voters, signer);
+        };
```

Comment:
I'm not sure we want to do this, because it gives implicit privilege to the address that ran the transaction which is not necessarily a given? I may legitimately want to use a burner account with no privileges to run the publish and then hand over the privilege to upgrade it to some other set of addresses, not including itself.

Reply:
yes this code would imply that the signer is registering themselves. To be honest the idea here was that the library owner would always have a right to vote for an upgrade if they were not the proposer. 
One could claim that if they want they should be passing their address in, and I may be fine with it, even like it more actually. Maybe it's just a question of educate people to the implication of one or the other.
This was trying to create the strong correlation between the owner of the cap and the signer which does not have to be there if we think it's counterintuitive (and it may be)

Reply:
I changed that and it makes the code cleaner, thanks

Reply:
resolving as I changed as suggested